parse/lex.go

```diff
@@ -111,6 +118,23 @@ func (s *state) cut(t TokenType) Token {
 	return token
 }
 
+// next returns the next rune in the input.
+func (s *state) next() rune {
+	if int(s.pos) >= len(s.input) {
+		s.width = 0
```

Comment:
Doesn't it need it initialize `s.pos`??

Reply:
int types are initialized to `zero`. So it is fine

Reply:
Oh, I understood.